Content.Server/_Moffstation/Speech/LongSpeechSystem.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Server.Speech;
+using Content.Shared.Speech;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server._Moffstation.Speech;
+
+public sealed class LongSpeechSystem : EntitySystem
+{
+    [Dependency] private readonly SpeechSoundSystem _speechSound = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<SpeechComponent, LongSpeechComponent>();
+        while (query.MoveNext(out var uid, out _, out var longSpeech))
+        {
+            if (longSpeech.NextSpeak > _gameTiming.CurTime)
+                continue;
+
+            if (longSpeech.WordCount <= 0)
+            {
+                RemCompDeferred<LongSpeechComponent>(uid);
+                continue;
+            }
+
+            longSpeech.WordCount--;
+            longSpeech.NextSpeak = _gameTiming.CurTime + longSpeech.Cooldown;
+            _audio.PlayPvs(longSpeech.Sound, uid, longSpeech.Params);
+        }
+    }
+
+    public void SpeakSentence(Entity<SpeechComponent> ent, string message)
+    {
+        //If we're already speaking, finish the current sentence.
+        if (HasComp<LongSpeechComponent>(ent))
+            return;
+
+        var longSpeech = EnsureComp<LongSpeechComponent>(ent);
+
+        if (_speechSound.GetSpeechSound(ent, message) is not { } sound)
+        {
+            RemCompDeferred<LongSpeechComponent>(ent);
+            return;
+        }
```

Comment:
```suggestion
        if (_speechSound.GetSpeechSound(ent, message) is not { } sound)
            return;

        var longSpeech = EnsureComp<LongSpeechComponent>(ent);
```
No need to add it to just remove it later, right?